include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -2429,6 +2429,60 @@ DataOut_DoFData<dim, patch_dim, spacedim, patch_spacedim>::
             Assert(false, ExcNotImplemented());
         }
 
+  // Automatically merge components which are located consequently and have the
+  // same name into a vector or a tensor.
+  if (patch_spacedim > 1)
+    {
+      const auto names = get_dataset_names();
+
+      for (unsigned int component_start = 0; component_start < names.size();)
+        {
+          DataComponentInterpretation::DataComponentInterpretation
+            component_interpretation =
+              DataComponentInterpretation::component_is_scalar;
+
+          unsigned int component_end = component_start + 1;
+          for (; component_end < names.size(); ++component_end)
+            if (names[component_end] != names[component_start])
+              break;
+
+          const unsigned int n_components = component_end - component_start;
+
+          if (n_components == patch_spacedim)
+            component_interpretation =
+              DataComponentInterpretation::component_is_part_of_vector;
+          else if (n_components == patch_spacedim * patch_spacedim)
+            component_interpretation =
+              DataComponentInterpretation::component_is_part_of_tensor;
```

Comment:
If there is no strong technical reason for it then I  would advocate to dispense of the `n_components == patch_spacedim` or `n_components == patch_spacedim * patch_spacedim` requirement and simple always put it into a vector (of arbitrary dimension).

One reason being that I can have a vector-valued output that is not exactly spacedim large. For example, a state vector for a conservation law, or a 2-dimensional surface vector that lives on a co-dimension 1 manifold.

The other, "spacedim * spacedim" isn't really a vector either but almost surely a rank-2 tensor.

So why not simply identifying everything that has the same name as being a vector?

Reply:
*blerg* And here I am complaining about that `DataComponentInterpretation` is broken.

Reply:
I believe that the code as written is correct. In the end, what matters is what the output format writers consider a "vector" and "tensor". And they think of a vector as something that has exactly `spacedim` components.